src/main/java/it/unisa/Amigo/repository/services/RepositoryServiceImpl.java

```diff
@@ -1,4 +1,82 @@
 package it.unisa.Amigo.repository.services;
 
-public class RepositoryServiceImpl {
+import it.unisa.Amigo.documento.domain.Documento;
+import it.unisa.Amigo.documento.service.DocumentoService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.transaction.Transactional;
+import java.util.List;
+
+/**
+ * Questa classe implementa i metodi per la logica di Business del sottosistema "Repository"
+ */
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class RepositoryServiceImpl implements RepositoryService {
+
+    private final DocumentoService documentoService;
+
+    /**
+     * Aggiunge un documento @{@link Documento} alla repository.
+     *
+     * @param file da aggiungere alla repository.
+     * @return true se il documento è stato aggiunto alla repository.
+     */
+    @Override
+    public boolean addDocumentoInRepository(MultipartFile file) {
+        if(file.getSize()==0 || file.getSize()>10485760){
```

Comment:
Usare delle costanti valide in tutta la classe per i limiti al formato e alle dimensioni del documento

Reply:
Quindi creiamo una classe che contiene tutte queste cose ?